platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java

```diff
@@ -179,6 +191,27 @@ private ImportMigrationConfigurationAdminEntry updateNecessaryConfigurationAdmin
     if (entry.getPid().equals("org.codice.ddf.security.filter.login.Session")) {
       entry.setPid("ddf.security.http.impl.HttpSessionFactory");
     }
+    // this configuration was updated in version 2.13.6
```

Comment:
✏️ This method is starting to be a trash can of `if`s. Should we start thinking of just keeping a list of filters (i.e. `List<Function<ImportMigrationConfigurationAdminEntry , ImportMigrationConfigurationAdminEntry>>` that this method goes through, passing the `entry` as a parameter to each one of them and having them return an entry as a result? That way we could register one such filter for the above `if` at line 191 and one for line 196 by moving the `if` statement in the private method and doing `filters.add(this:: updateMetacardValidityFilterPluginConfiguration)` in addition to adding one of the above `if` as part of the construction of this migratable

Reply:
I referenced [this](https://github.com/codice/ddf/blob/master/platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java#L205) when writing this. 
This function itself is being called from within a filter [here](https://github.com/codice/ddf/blob/master/platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java#L187). What do you think would be the best way to structure this?

Reply:
I went back to using if statements because I need to pass in the context to get the imported hostname and I saw that Function can only be functions with one parameter. If there's a better way to do this, please let me know!

Reply:
`java.util.function.BiFunction`